microsoft_atp/komand_microsoft_atp/connection/connection.py

```diff
@@ -1,424 +1,314 @@
+from .schema import ConnectionSchema, Input
 import komand
+# Custom imports below
+from komand.exceptions import ConnectionTestException, PluginException
 import json
 import requests
-import maya
-from .schema import ConnectionSchema
-from komand.exceptions import ConnectionTestException
-
-
-# Custom imports below
+import time
 
 
 class Connection(komand.Connection):
 
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
-        self.start_from_payload = {"sinceTimeUtc": "2010-01-01T00:00:00.000"}
-        self.api_version = "/api"
-        self.alert_endpoint = "/Alerts"
-        self.machine_endpoint = "/machines"
         self.session = requests.Session()
-        self.seconds_ago = 0
-        self.current_time = maya.now()
-        self.host_url = ""
-        self.auth_url = ""
-        self.client_id = ""
-        self.client_secret = ""
         self.resource_url = "https://api.securitycenter.windows.com"
 
+
     def connect(self, params):
         self.logger.info("Connecting...")
 
-        self.seconds_ago = params.get("seconds_ago", 0)
-        self.current_time = maya.now().add(seconds=-int(self.seconds_ago))
+        self.app_id = params.get(Input.APPLICATION_ID)
+        self.app_secret = params.get(Input.APPLICATION_SECRET).get("secretKey")
+        self.tenant = params.get(Input.DIRECTORY_ID)
 
-        self.host_url = params.get("host")
+        self.auth_url = f"https://login.windows.net/{self.tenant}/oauth2/token"
 
-        self.auth_url = params.get("auth_url")
-        self.client_id = params.get("client_id")
-        self.client_secret = params.get("client_secret")
+        self.api_token = ""
 
-        self.resource_url = params.get("resource_url")
+        # Auth tokens expire after 1 hour. Only make that call if we need to
+        self.time_ago = 0  # Jan 1, 1970
+        self.time_now = time.time()  # More than 1 hour since 1978
 
+        self.check_and_refresh_api_token()
         self.logger.info("Setup Complete")
 
-    def update_access_token(self):
-        self.logger.info("Updating Access Token...")
-        o_auth_token = self.get_access_token(self.auth_url, self.client_id, self.client_secret)
+    def check_and_refresh_api_token(self, force_refresh_token=False):
```

Comment:
Can you move the non-connect functions into an API class?

Reply:
Yes: 

https://issues.corp.rapid7.com/browse/SOAR-1830